### ISSUE TITLE
Don't search lyrics on Internet when offline. 

### ISSUE THREAD

**Issue by Rexadev** [jacquesh/foo_openlyrics#188]
> - How you think the new feature/improvement would be helpful, or what problem you think it would solve

Not count against" lyrics not found will not auto search"
Save CPU and Battery

**Comment by jacquesh** [comment]
Good idea, although off the top of my head I'm not entirely sure how to characterise "when offline". Windows has some mechanism for determining this so we may be able to hook into that. Will require some investigation.

**Comment by Rexadev** [comment]
> how to characterise "when offline". Windows has some mechanism for determining this so we may be able to hook into that. Will require some investigation.

Ask peter to include it in v2

**Comment by Rexadev** [comment]
`
diff --git a/src/openlyrics.cpp b/src/openlyrics.cpp
index 6c34887..1f3d3f2 100644
--- a/src/openlyrics.cpp
+++ b/src/openlyrics.cpp
@@ -399,6 +399,10 @@ class foo_openlyrics : public lyrics_provider_v3 {
         return LYRICS_OK;
     }
 
+    if (!m_internet_access) {
+        return LYRICS_OFFLINE;
+    }
+
     // Start a search on the Internet
     m_internet_searching = true;
     static_api_ptr_t<lyricwiki_search_manager_v2>()->search_lyrics_async(m_song_info, m_abort_callback, search_callback(m_lyrics_search_done_callback));
@@ -423,6 +427,7 @@ class foo_openlyrics : public lyrics_provider_v3 {
         m_internet_access = false;
         break;
     }
+    update_ui();
     return LYRICS_OK;
 }
 
@@ -1293,6 +1298,10 @@ class foo_openlyrics : public lyrics_provider_v3 {
             return LYRICS_OK;
         }
 
+        if (!m_internet_access) {
+            return LYRICS_OFFLINE;
+        }
+
         // Start a search on the Internet
         m_internet_searching = true;
         static_api_ptr_t<openlyrics_search_manager>()->search_lyrics_async(m_song_info, m_abort_callback, search_callback(m_lyrics_search_done_callback));
@@ -1317,6 +1326,7 @@ class foo_openlyrics : public lyrics_provider_v3 {
             m_internet_access = false;
             break;
         }
+        update_ui();
         return LYRICS_OK;
     }
 
@@ -1827,6 +1837,10 @@ class foo_openlyrics : public lyrics_provider_v3 {
         return LYRICS_OK;
     }
 
+    if (!m_internet_access) {
+        return LYRICS_OFFLINE;
+    }
+
     // Start a search on the Internet
     m_internet_searching = true;
     static_api_ptr_t<azlyrics_search_manager>()->search_lyrics_async(m_song_info, m_abort_callback, search_callback(m_lyrics_search_done_callback));
@@ -1851,6 +1865,7 @@ class foo_openlyrics : public lyrics_provider_v3 {
         m_internet_access = false;
         break;
     }
+    update_ui();
     return LYRICS_OK;
 }
 
@@ -2164,6 +2179,7 @@ class foo_openlyrics : public lyrics_provider_v3 {
         // update the state
         m_internet_access = true;
         m_internet_searching = false;
+        update_ui();
     }
 };
 
@@ -2187,6 +2203,7 @@ class foo_openlyrics : public lyrics_provider_v3 {
         // update the state
         m_internet_access = true;
         m_internet_searching = false;
+        update_ui();
     }
 };
 

`

**Comment by Rexadev** [comment]
```
diff --git a/src/openlyrics.cpp b/src/openlyrics.cpp
index 6c34887..1f3d3f2 100644
--- a/src/openlyrics.cpp
+++ b/src/openlyrics.cpp
@@ -399,6 +399,10 @@ class foo_openlyrics : public lyrics_provider_v3 {
         return LYRICS_OK;
     }
 
+    if (!m_internet_access) {
+        return LYRICS_OFFLINE;
+    }
+
     // Start a search on the Internet
     m_internet_searching = true;
     static_api_ptr_t<lyricwiki_search_manager_v2>()->search_lyrics_async(m_song_info, m_abort_callback, search_callback(m_lyrics_search_done_callback));
@@ -423,6 +427,7 @@ class foo_openlyrics : public lyrics_provider_v3 {
         m_internet_access = false;
         break;
     }
+    update_ui();
     return LYRICS_OK;
 }
 
@@ -1293,6 +1298,10 @@ class foo_openlyrics : public lyrics_provider_v3 {
             return LYRICS_OK;
         }
 
+        if (!m_internet_access) {
+            return LYRICS_OFFLINE;
+        }
+
         // Start a search on the Internet
         m_internet_searching = true;
         static_api_ptr_t<openlyrics_search_manager>()->search_lyrics_async(m_song_info, m_abort_callback, search_callback(m_lyrics_search_done_callback));
@@ -1317,6 +1326,7 @@ class foo_openlyrics : public lyrics_provider_v3 {
             m_internet_access = false;
             break;
         }
+        update_ui();
         return LYRICS_OK;
     }
 
@@ -1827,6 +1837,10 @@ class foo_openlyrics : public lyrics_provider_v3 {
         return LYRICS_OK;
     }
 
+    if (!m_internet_access) {
+        return LYRICS_OFFLINE;
+    }
+
     // Start a search on the Internet
     m_internet_searching = true;
     static_api_ptr_t<azlyrics_search_manager>()->search_lyrics_async(m_song_info, m_abort_callback, search_callback(m_lyrics_search_done_callback));
@@ -1851,6 +1865,7 @@ class foo_openlyrics : public lyrics_provider_v3 {
         m_internet_access = false;
         break;
     }
+    update_ui();
     return LYRICS_OK;
 }
 
@@ -2164,6 +2179,7 @@ class foo_openlyrics : public lyrics_provider_v3 {
         // update the state
         m_internet_access = true;
         m_internet_searching = false;
+        update_ui();
     }
 };
 
@@ -2187,6 +2203,7 @@ class foo_openlyrics : public lyrics_provider_v3 {
         // update the state
         m_internet_access = true;
         m_internet_searching = false;
+        update_ui();
     }
 };
 

```

**Comment by Rexadev** [comment]
> Good idea, although off the top of my head I'm not entirely sure how to characterise "when offline". Windows has some mechanism for determining this so we may be able to hook into that. Will require some investigation.

Just Ping 1.1.1.1 or 9.9.9.9 on start up and every 2 minutes then

**Comment by Rexadev** [comment]
@marc2k3 will you please help with this ?

**Comment by marc2k3** [comment]
No idea why I'm being pinged on this, But I searched and found this component is using `http_client` from the fb2k SDK exactly like I do. I know from experience it takes around 10 seconds to timeout when offline. So what? I've lived with it for years and so have many others. It's a big nothing burger.

If people want to re-invent the wheel to get an "instant" fail, that's on them.

**Comment by Rexadev** [comment]
@marc2k3 sorry for pinging you I saw you sharing your f2k dev experience on the component repo so i thought I should ask you. if you say i would ping you again.
